app/services/spam_report_service.rb

```diff
@@ -1,14 +1,29 @@
 class SpamReportService < ApplicationService
-  attr_reader :delivery_attempt
+  attr_reader :delivery_attempt_id, :email_address
 
-  def initialize(delivery_attempt)
-    @delivery_attempt = delivery_attempt
+  def initialize(delivery_attempt_id, email_address)
+    @delivery_attempt_id = delivery_attempt_id
+    @email_address = email_address
   end
 
   def call
-    subscriber_id = delivery_attempt.email.subscriber_id
-    subscriber = Subscriber.find(subscriber_id)
+    return if email.marked_as_spam?
+
+    UnsubscribeAllService.call(subscriber, :marked_as_spam)
```

Comment:
This call doesn't look safe as subscriber could be nil and I don't think this service anticipates it to be nil.